core/modules/simpletest/tests/error_test.module

```diff
@@ -40,9 +40,13 @@ function error_test_generate_warnings($collect_errors = FALSE) {
   // Tell Backdrop error reporter to send errors to Simpletest or not.
   define('SIMPLETEST_COLLECT_ERRORS', $collect_errors);
   // This will generate a notice.
-  $monkey_love = $bananas;
+  $notice = new \stdClass();
+  $notice == 1 ? 1 : 0;
   // This will generate a warning.
-  $awesomely_big = 1/0;
+  $a = '';
+  foreach ($a as $b) {
+
+  }
```

Comment:
I don't get what we're doing here 🤔

Reply:
This is a pot from Drupal: https://www.drupal.org/project/drupal/issues/3206429

We're intentionally causing PHP notices and warnings. A large number of notices were "upgraded" to warnings in PHP 8, and others were upgraded from warnings to errors (like division by 0). This is one of the few ways we can cause a warning in all versions of PHP.

Reply:
Thanks @quicksketch ...worth putting that in an inline comment, so that the next person such as myself reading this bit of code doesn't wonder why?

Reply:
...also, if we are cross-porting D7 changes, we're missing things. See my PR https://github.com/hosef/backdrop/pull/1 (@hosef might need to review and merge)

Reply:
Thanks @klonos! I can't merge PRs on @hosef's repository, but I can pull the commit into this PR. I've done so and your changes are now in as the next commit (https://github.com/backdrop/backdrop/pull/3412/commits/875253dfb6464e33b6d93d5796238a27da68649d).